autoscale/src/main/java/com/sequenceiq/periscope/monitor/evaluator/flow/FlowEvaluator.java

```diff
@@ -0,0 +1,89 @@
+package com.sequenceiq.periscope.monitor.evaluator.flow;
+
+import java.time.Instant;
+import java.util.List;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.periscope.domain.Cluster;
+import com.sequenceiq.periscope.monitor.context.ClusterIdEvaluatorContext;
+import com.sequenceiq.periscope.monitor.context.EvaluatorContext;
+import com.sequenceiq.periscope.monitor.evaluator.EvaluatorExecutor;
+import com.sequenceiq.periscope.monitor.evaluator.EventPublisher;
+import com.sequenceiq.periscope.monitor.event.ScalingTriggerCleanupEvent;
+import com.sequenceiq.periscope.monitor.event.ScalingTriggerFlowStatusEvent;
+import com.sequenceiq.periscope.service.ClusterService;
+import com.sequenceiq.periscope.service.ScalingTriggerService;
+
+@Component("FlowEvaluator")
+@Scope("prototype")
+public class FlowEvaluator extends EvaluatorExecutor {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FlowEvaluator.class);
+
+    private static final String EVALUATOR_NAME = FlowEvaluator.class.getName();
+
+    private long clusterId;
+
+    @Value("${periscope.scalingTrigger.cleanup-duration.minutes}")
+    private Long cleanupDuration;
+
+    @Inject
+    private ClusterService clusterService;
+
+    @Inject
+    private ScalingTriggerService scalingTriggerService;
+
+    @Inject
+    private EventPublisher eventPublisher;
+
+    @Nonnull
+    @Override
+    public EvaluatorContext getContext() {
+        return new ClusterIdEvaluatorContext(clusterId);
+    }
+
+    @Override
+    public String getName() {
+        return EVALUATOR_NAME;
+    }
+
+    @Override
+    public void setContext(EvaluatorContext context) {
+        this.clusterId = (long) context.getData();
+    }
+
+    @Override
+    protected void execute() {
+        LOGGER.info("Executing FlowEvaluator for cluster: {}", clusterId);
+        String stackCrn = null;
+        long start = Instant.now().toEpochMilli();
+        try {
+            Cluster cluster = clusterService.findById(clusterId);
+            stackCrn = cluster.getStackCrn();
+
+            List<Long> scalingTriggersOlderThanDuration = scalingTriggerService.findAllIdsOlderThanDurationForCluster(cluster, cleanupDuration);
+            List<Long> successfulScalingTriggers = scalingTriggerService.findAllIdsOfSuccessfulScalingTriggersForCluster(cluster);
+
+            if (!scalingTriggersOlderThanDuration.isEmpty()) {
+                LOGGER.info("Sending scalingTrigger cleanup event for cluster: {}", stackCrn);
+                eventPublisher.publishEvent(new ScalingTriggerCleanupEvent(clusterId, cleanupDuration, scalingTriggersOlderThanDuration));
+            } else if (!successfulScalingTriggers.isEmpty()) {
```

Comment:
Not an else if ?
We need to update trigger statuses even if there are cleanup events to be performed.

Reply:
Refactored this entirely